projects/pastanaga/src/lib/calendar/date-input.component.ts

```diff
@@ -0,0 +1,102 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, Input, Output, Renderer2 } from '@angular/core';
+import { PopupComponent } from '../popup/popup.component';
+import { PopupService } from '../popup/popup.service';
+import { markForCheck } from '../common/utils';
+
+@Component({
+    selector: 'pa-date-input',
+    templateUrl: './date-input.component.html',
+    styleUrls: ['../popup/_popup.scss', './date-input.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DateInputComponent extends PopupComponent {
+
+    dateInput = '';
+    dateError = false;
+    isValidDate = false;
+    currentDate: Date = new Date();
+    focused = false;
+
+    @Input() dateHelp: string = '';
+    @Input() datePlaceholder: string = 'mm/dd/yyyy';
+
+    @Output() select: EventEmitter<Date | null> = new EventEmitter<Date>();
+
+    constructor(
+        public popupService: PopupService,
+        public renderer: Renderer2,
+        public element: ElementRef,
+        public cdr: ChangeDetectorRef,
+    ) {
+        super(popupService, renderer, element, cdr);
+    }
+
+    selectDate(date: Date) {
+        this.currentDate = date;
+        this.dateInput = '';
+        const dd = date.getDate();
+        const mm = date.getMonth() + 1;
+        const yyyy = date.getFullYear();
+        const today = new Date();
+        const yesterday = new Date();
+        yesterday.setDate(yesterday.getDate() - 1);
+        if (today.getFullYear() === date.getFullYear()){
+            if (today.getMonth() === date.getMonth() && today.getDate() === date.getDate()){
+                this.dateInput = 'Today';
+            } else if (yesterday.getMonth() === date.getMonth() && yesterday.getDate() === date.getDate()){
+                this.dateInput = 'Yesterday';
+            }
+        }
+        if (this.dateInput === '') {
+            this.dateInput = mm + '/' + dd + '/' + yyyy; // American date format
+        }
```

Comment:
use `date-fns` helper methods instead.
Here the doc: https://date-fns.org/v2.9.0/docs/